scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+    return output
+
+
+@app.route('/{}/service/logs'.format(CLUSTER_API), methods=['POST'])
+def service_logs():
+    """
+    Web call to the logs of a service
+    :request_param service: the name of the service
+    :request_param lines: total lines | if omitted default value 10 will be used
+    """
+    lines = 10
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    if "lines" in request.json:
+        lines = request.json["lines"]
+    # more info on logs could be obtained by using: --output=json-pretty
+    output = subprocess.check_output("journalctl --lines={} --unit=snap.{}.service --no-pager".format(lines, request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/addon/enable'.format(CLUSTER_API), methods=['POST'])
+def enable():
+    """
+    Web call to microk8s.enable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-enable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/addon/disable'.format(CLUSTER_API), methods=['POST'])
+def disable():
+    """
+    Web call to microk8s.disable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-disable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/start'.format(CLUSTER_API), methods=['POST'])
+def start():
+    """
+    Web call for microk8s.start
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-start.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/stop'.format(CLUSTER_API), methods=['POST'])
+def stop():
+    """
+    Web call for microk8s.stop
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-stop.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/overview'.format(CLUSTER_API), methods=['POST'])
```

Comment:
Getting access tot he cluster like this is not something we recommend. If you want to have an overview of the cluster you can query the cluster's API server directly.

Reply:
The cluster agent API is not meant to be a proxy for the k8s API server.

MicroDash runs as a pod in MicroK8s and thus should access to the cluster API server as described in the official docs [1]. This way we comply with the authn & authz policies set by the cluster administrator.

[1] https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#accessing-the-api-from-within-a-pod